code/__HELPERS/type2type.dm

```diff
@@ -162,10 +162,14 @@
 //Converts a string into a list by splitting the string at each delimiter found. (discarding the seperator)
 /proc/text2list(text, delimiter="\n")
 	var/delim_len = length(delimiter)
-	if(delim_len < 1) return list(text)
 	. = list()
 	var/last_found = 1
-	var/found
+	var/found = 1
+	if(delim_len < 1)
+		var/text_len = length(text)
+		while(found++ <= text_len)
+			. += copytext(text,found-1, found)
+		return .

```

Comment:
please no

can't you put the ```do``` section under an ```else```? allowing you to avoid the horrors of ```return . ```